rust/private/rust_analyzer.bzl

```diff
@@ -229,7 +229,7 @@ def _create_single_crate(ctx, info):
     crate["root_module"] = path_prefix + info.crate.root.path
     crate_root = path_prefix + info.crate.root.dirname
 
-    if info.build_info != None:
+    if info.build_info != None and info.build_info.out_dir != None:
```

Comment:
> Should you change the PR title to "Add CI Jobs for rust-analyzer rule" instead? I don't see any fix in the PR.

This is the fix. Though I did open this PR without it to see what CI would do. But I was faster than CI locally so made a few changes and then added the proper fix

Reply:
Looking at the error, do you have an idea why `out_dir` is `None` for that particular target?

```
ERROR: /Users/user/Code/rules_rust/examples/bindgen/BUILD.bazel:4:21: in @@rules_rust//rust/private:rust_analyzer.bzl%rust_analyzer_aspect aspect on 
```

Reply:
@daivinhtran yeah, this is intentionally `None` per the implementation of `rust_bindgen`
https://github.com/bazelbuild/rules_rust/blob/0.38.0/bindgen/private/bindgen.bzl#L157